examples/hosts/app-integration/external-data/src/view/taskListView.tsx

```diff
@@ -6,9 +6,10 @@ import { CollaborativeInput } from "@fluid-experimental/react-inputs";
 
 import React, { useCallback, useEffect, useRef, useState } from "react";
 
-import type { ExternalSnapshotTask, ITask, ITaskList } from "../model-interface";
+import type { ExternalSnapshotTask, IAppModel, ITask, ITaskList } from "../model-interface";
 
 interface ITaskRowProps {
+	readonly model: IAppModel;
```

Comment:
I don't think we need to pass the whole model. Passing a string (the "leader" property) should be enough.

I'd also say we should probably make it something like "leaderClientId" so it can identify who is the leader. At the "bottom" of the component tree, where we need to decide if we should display the button or not, we can compare against the current client's id (which might also need to be passed somehow) to determine if the current client is the leader.